ToolBox/IDataConverter.cs

```diff
@@ -0,0 +1,125 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Auxiliary;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace QuantConnect.ToolBox
+{
+    /// <summary>
+    /// Interface defines common patterns that are involved in implementing a ToolBox converter
+    /// for custom data sources. We provide many generic implementable methods so that your converter
+    /// is capable of processing multiple data sources per instance
+    public interface IDataConverter
+    {
+        /// <summary>
+        /// Responsible for handling discovery of map files that deal with
+        /// mapping events (e.g. delistings, renames, listing after delisting, etc.)
+        /// </summary>
+        MapFileResolver MapFileResolver { get; }
```

Comment:
Having a MapFileResolver as a definition in IDataConverter sounds like a loss of generality, it is focused in USA equity markets.

Reply:
I think you're right. Seeing this with a fresh mind, I think I need to reconsider the addition of this.

In general, I think this "abstraction" needs more thought, since trying to shoehorn in mapping as is is not a solid design pattern, especially considering that what you've said holds a lot of truth.